### PR TITLE
Bump LTS version + test nightly LTS.

### PR DESCRIPTION
**Pull request by ryzhyk**
- Bumped LTS version to 16.21.
- Tried compiling DDlog with nightly LTS based on GHC 8.10.2 and fixed warnings produced by the new compiler.
- Added a commented out line to `stack.yml` to enable nightly LTS.

@Kixiron

### CODE REVIEW

**Comment by Kixiron** [comment]
It does really really well, but it seems ddlog depends on `regix-postix`, which tries to compile some header files that are assumedly linux-only

Relevant portion:

```
regex-posix          > compiling .stack-work\dist\a3a5fe88\build\Text\Regex\Posix\Wrap_hsc_make.c failed (exit code 1)
regex-posix          > rsp file was: ".stack-work\\dist\\a3a5fe88\\build\\Text\\Regex\\Posix\\hsc4BE6.rsp"
regex-posix          > command was: C:\Users\Chase\AppData\Local\Programs\stack\x86_64-windows\ghc-8.10.2\lib\../mingw/bin\gcc.exe -c .stack-work\dist\a3a5fe88\build\Text\Regex\Posix\Wrap_hsc_make.c -o .stack-work\dist\a3a5fe88\build\Text\Regex\Posix\Wrap_hsc_make.o -D__GLASGOW_HASKELL__=810 -Dmingw32_BUILD_OS=1 -Dx86_64_BUILD_ARCH=1 -Dmingw32_HOST_OS=1 -Dx86_64_HOST_ARCH=1 -Icbits -IC:\Users\Chase\AppData\Local\Programs\stack\x86_64-windows\msys2-20180531\mingw64\include -I.stack-work\dist\a3a5fe88\build\cbits -IC:\Users\Chase\AppData\Local\Programs\stack\x86_64-windows\msys2-20180531\mingw64\include -I.stack-work\dist\a3a5fe88\build\autogen -I.stack-work\dist\a3a5fe88\build\global-autogen -include .stack-work\dist\a3a5fe88\build\autogen\cabal_macros.h -IC:\Users\Chase\AppData\Local\Programs\stack\x86_64-windows\msys2-20180531\mingw64\include -IC:\Users\Chase\AppData\Local\Programs\stack\x86_64-windows\ghc-8.10.2\lib\bytestring-0.10.10.0\include -IC:\Users\Chase\AppData\Local\Programs\stack\x86_64-windows\ghc-8.10.2\lib\base-4.14.1.0\include -IC:\Users\Chase\AppData\Local\Programs\stack\x86_64-windows\ghc-8.10.2\lib\integer-gmp-1.0.3.0\include -IC:\Users\Chase\AppData\Local\Programs\stack\x86_64-windows\ghc-8.10.2\lib/include -IC:\Users\Chase\AppData\Local\Programs\stack\x86_64-windows\ghc-8.10.2/lib/include/
regex-posix          > error: Wrap.hsc:96:10: fatal error: regex.h: No such file or directory
regex-posix          > compilation terminated.
```

<details>
<summary>Full output</summary>

```
❯ stack build
WARNING: Ignoring regex-posix's bounds on base (<0 && >=4.3 && <4.15); using base-4.14.1.0.
Reason: allow-newer enabled.
OneTuple             > using precompiled package
base-compat          > using precompiled package
base-orphans         > using precompiled package
cabal-doctest        > using precompiled package
clock                > using precompiled package
colour               > using precompiled package
dlist                > using precompiled package
fgl                  > using precompiled package
file-embed           > using precompiled package
hashable             > using precompiled package
network              > using precompiled package
old-locale           > using precompiled package
polyparse            > using precompiled package
primitive            > using precompiled package
random               > using precompiled package
raw-strings-qq       > using precompiled package
regex-base           > using precompiled package
split                > using precompiled package
tagged               > using precompiled package
transformers-compat  > using precompiled package
githash              > configure
Progress 1/51: base-compat, base-orphans, cabal-doctest, clock, colour, dlist, fgl, file-embed, githash, hashable, integer-logarithms, network, old-locale, polyparse, primitive, random, raw-st
tuple                > using precompiled package
Progress 1/51: base-compat, base-orphans, cabal-doctest, clock, colour, dlist, fgl, file-embed, githash, hashable, integer-logarithms, network, old-locale, polyparse, primitive, random, raw-st
base-compat-batteries> using precompiled package
Progress 1/51: base-compat, base-orphans, cabal-doctest, clock, colour, dlist, fgl, file-embed, githash, hashable, integer-logarithms, network, old-locale, polyparse, primitive, random, raw-st
Progress 2/51: base-compat-batteries, base-orphans, cabal-doctest, clock, colour, dlist, fgl, file-embed, githash, hashable, integer-logarithms, network, old-locale, polyparse, primitive, rand
wl-pprint-text       > using precompiled package
Progress 2/51: base-compat-batteries, base-orphans, cabal-doctest, clock, colour, dlist, fgl, file-embed, githash, hashable, integer-logarithms, network, old-locale, polyparse, primitive, rand
Progress 3/51: base-compat-batteries, cabal-doctest, clock, colour, dlist, fgl, file-embed, githash, hashable, integer-logarithms, network, old-locale, polyparse, primitive, random, raw-string
Progress 4/51: base-compat-batteries, clock, colour, dlist, fgl, file-embed, githash, hashable, integer-logarithms, network, old-locale, polyparse, primitive, random, raw-strings-qq, regex-bas
Progress 5/51: base-compat-batteries, colour, dlist, extra, fgl, file-embed, githash, hashable, integer-logarithms, network, old-locale, polyparse, primitive, random, raw-strings-qq, regex-bas
extra                > using precompiled package
Progress 5/51: base-compat-batteries, colour, dlist, extra, fgl, file-embed, githash, hashable, integer-logarithms, network, old-locale, polyparse, primitive, random, raw-strings-qq, regex-bas
Progress 6/51: base-compat-batteries, dlist, extra, fgl, file-embed, githash, hashable, integer-logarithms, network, old-locale, polyparse, primitive, random, raw-strings-qq, regex-base, split
Progress 7/51: base-compat-batteries, extra, fgl, file-embed, githash, hashable, integer-logarithms, network, old-locale, polyparse, primitive, random, raw-strings-qq, regex-base, split, tagge
Progress 8/51: base-compat-batteries, extra, file-embed, githash, hashable, integer-logarithms, network, old-locale, polyparse, primitive, random, raw-strings-qq, regex-base, split, tagged, th
githash              > Configuring githash-0.1.5.0...
Progress 8/51: base-compat-batteries, extra, file-embed, githash, hashable, integer-logarithms, network, old-locale, polyparse, primitive, random, raw-strings-qq, regex-base, split, tagged, th
Progress 9/51: base-compat-batteries, extra, githash, hashable, integer-logarithms, network, old-locale, polyparse, primitive, random, raw-strings-qq, regex-base, split, tagged, th-abstraction
Progress 10/51: base-compat-batteries, data-fix, extra, githash, integer-logarithms, network, old-locale, polyparse, primitive, random, raw-strings-qq, regex-base, split, tagged, th-abstractio
data-fix             > using precompiled package
Progress 10/51: base-compat-batteries, data-fix, extra, githash, integer-logarithms, network, old-locale, polyparse, primitive, random, raw-strings-qq, regex-base, split, tagged, th-abstractio
unordered-containers > using precompiled package
Progress 10/51: base-compat-batteries, data-fix, extra, githash, integer-logarithms, network, old-locale, polyparse, primitive, random, raw-strings-qq, regex-base, split, tagged, th-abstractio
Progress 11/51: base-compat-batteries, data-fix, extra, githash, integer-logarithms, network-bsd, old-locale, polyparse, primitive, random, raw-strings-qq, regex-base, split, tagged, th-abstra
network-bsd          > using precompiled package
Progress 11/51: base-compat-batteries, data-fix, extra, githash, integer-logarithms, network-bsd, old-locale, polyparse, primitive, random, raw-strings-qq, regex-base, split, tagged, th-abstra
githash              > build
Progress 11/51: base-compat-batteries, data-fix, extra, githash, integer-logarithms, network-bsd, old-locale, polyparse, primitive, random, raw-strings-qq, regex-base, split, tagged, th-abstra
Progress 12/51: base-compat-batteries, data-fix, extra, githash, integer-logarithms, network-bsd, old-time, polyparse, primitive, random, raw-strings-qq, regex-base, split, tagged, th-abstract
old-time             > using precompiled package
Progress 12/51: base-compat-batteries, data-fix, extra, githash, integer-logarithms, network-bsd, old-time, polyparse, primitive, random, raw-strings-qq, regex-base, split, tagged, th-abstract
githash              > Preprocessing library for githash-0.1.5.0..
Progress 12/51: base-compat-batteries, data-fix, extra, githash, integer-logarithms, network-bsd, old-time, polyparse, primitive, random, raw-strings-qq, regex-base, split, tagged, th-abstract
githash              > Building library for githash-0.1.5.0..
Progress 12/51: base-compat-batteries, data-fix, extra, githash, integer-logarithms, network-bsd, old-time, polyparse, primitive, random, raw-strings-qq, regex-base, split, tagged, th-abstract
githash              > [1 of 2] Compiling GitHash
Progress 12/51: base-compat-batteries, data-fix, extra, githash, integer-logarithms, network-bsd, old-time, polyparse, primitive, random, raw-strings-qq, regex-base, split, tagged, th-abstract
Progress 13/51: base-compat-batteries, data-fix, extra, githash, integer-logarithms, network-bsd, old-time, primitive, random, raw-strings-qq, regex-base, split, tagged, th-abstraction, time-c
Progress 14/51: base-compat-batteries, data-fix, extra, githash, integer-logarithms, network-bsd, old-time, random, raw-strings-qq, regex-base, split, tagged, th-abstraction, time-compat, tran
vector               > using precompiled package
Progress 14/51: base-compat-batteries, data-fix, extra, githash, integer-logarithms, network-bsd, old-time, random, raw-strings-qq, regex-base, split, tagged, th-abstraction, time-compat, tran
wide-word            > using precompiled package
Progress 14/51: base-compat-batteries, data-fix, extra, githash, integer-logarithms, network-bsd, old-time, random, raw-strings-qq, regex-base, split, tagged, th-abstraction, time-compat, tran
Progress 15/51: base-compat-batteries, data-fix, extra, githash, integer-logarithms, network-bsd, old-time, raw-strings-qq, regex-base, split, tagged, temporary, th-abstraction, time-compat, t
temporary            > using precompiled package
Progress 15/51: base-compat-batteries, data-fix, extra, githash, integer-logarithms, network-bsd, old-time, raw-strings-qq, regex-base, split, tagged, temporary, th-abstraction, time-compat, t
uuid-types           > using precompiled package
Progress 15/51: base-compat-batteries, data-fix, extra, githash, integer-logarithms, network-bsd, old-time, raw-strings-qq, regex-base, split, tagged, temporary, th-abstraction, time-compat, t
githash              > [2 of 2] Compiling Paths_githash
Progress 15/51: base-compat-batteries, data-fix, extra, githash, integer-logarithms, network-bsd, old-time, raw-strings-qq, regex-base, split, tagged, temporary, th-abstraction, time-compat, t
Progress 16/51: base-compat-batteries, data-fix, extra, githash, integer-logarithms, network-bsd, old-time, regex-base, split, tagged, temporary, th-abstraction, time-compat, transformers-comp
githash              > copy/register
Progress 16/51: base-compat-batteries, data-fix, extra, githash, integer-logarithms, network-bsd, old-time, regex-base, split, tagged, temporary, th-abstraction, time-compat, transformers-comp
githash              > Installing library in C:\sr\snapshots\6f4d65c6\lib\x86_64-windows-ghc-8.10.2\githash-0.1.5.0-IZKA1l7MzNzK5uWAxvMlqB
Progress 16/51: base-compat-batteries, data-fix, extra, githash, integer-logarithms, network-bsd, old-time, regex-base, split, tagged, temporary, th-abstraction, time-compat, transformers-comp
Progress 17/51: base-compat-batteries, data-fix, extra, githash, integer-logarithms, network-bsd, old-time, regex-posix, split, tagged, temporary, th-abstraction, time-compat, transformers-com
githash              > Registering library for githash-0.1.5.0..
Progress 17/51: base-compat-batteries, data-fix, extra, githash, integer-logarithms, network-bsd, old-time, regex-posix, split, tagged, temporary, th-abstraction, time-compat, transformers-com
Progress 18/51: base-compat-batteries, data-fix, extra, githash, integer-logarithms, network-bsd, old-time, regex-posix, tagged, temporary, th-abstraction, time-compat, transformers-compat, tu
Progress 19/51: base-compat-batteries, data-fix, distributive, extra, githash, integer-logarithms, network-bsd, old-time, regex-posix, temporary, th-abstraction, time-compat, transformers-comp
distributive         > using precompiled package
Progress 19/51: base-compat-batteries, data-fix, distributive, extra, githash, integer-logarithms, network-bsd, old-time, regex-posix, temporary, th-abstraction, time-compat, transformers-comp
Progress 20/51: base-compat-batteries, data-fix, distributive, extra, githash, integer-logarithms, network-bsd, old-time, regex-posix, temporary, th-abstraction, time-compat, tuple, unordered-
Progress 21/51: base-compat-batteries, data-fix, distributive, extra, githash, integer-logarithms, network-bsd, old-time, regex-posix, temporary, th-abstraction, time-compat, unordered-contain
Progress 22/51: data-fix, distributive, extra, githash, integer-logarithms, network-bsd, old-time, regex-posix, temporary, th-abstraction, time-compat, unordered-containers, uuid-types, vector
Progress 23/51: data-fix, distributive, extra, githash, integer-logarithms, network-bsd, old-time, regex-posix, temporary, th-abstraction, time-compat, unordered-containers, uuid-types, vector
Progress 24/51: data-fix, distributive, githash, integer-logarithms, network-bsd, old-time, regex-posix, temporary, th-abstraction, time-compat, unordered-containers, uuid-types, vector, wide-
integer-logarithms   > configure
hslogger             > using precompiled package
integer-logarithms   > Configuring integer-logarithms-1.0.3.1...
integer-logarithms   > build
integer-logarithms   > Preprocessing library for integer-logarithms-1.0.3.1..
integer-logarithms   > Building library for integer-logarithms-1.0.3.1..
integer-logarithms   > [1 of 4] Compiling GHC.Integer.Logarithms.Compat
integer-logarithms   > [2 of 4] Compiling Math.NumberTheory.Logarithms
graphviz             > using precompiled package
integer-logarithms   > [3 of 4] Compiling Math.NumberTheory.Powers.Integer
integer-logarithms   > [4 of 4] Compiling Math.NumberTheory.Powers.Natural
integer-logarithms   > copy/register
integer-logarithms   > Installing library in C:\sr\snapshots\6f4d65c6\lib\x86_64-windows-ghc-8.10.2\integer-logarithms-1.0.3.1-8l1ymRbE3rBG8A7FKIRWkt
integer-logarithms   > Registering library for integer-logarithms-1.0.3.1..
comonad              > using precompiled package
th-abstraction       > configure
th-abstraction       > Configuring th-abstraction-0.4.0.0...
th-abstraction       > build
th-abstraction       > Preprocessing library for th-abstraction-0.4.0.0..
th-abstraction       > Building library for th-abstraction-0.4.0.0..
time-compat          > configure
th-abstraction       > [1 of 3] Compiling Language.Haskell.TH.Datatype.Internal
th-abstraction       > [2 of 3] Compiling Language.Haskell.TH.Datatype.TyVarBndr
time-compat          > Configuring time-compat-1.9.4...
th-abstraction       > [3 of 3] Compiling Language.Haskell.TH.Datatype
time-compat          > build
regex-posix          > configure
time-compat          > Preprocessing library for time-compat-1.9.4..
time-compat          > Building library for time-compat-1.9.4..
regex-posix          > Configuring regex-posix-0.96.0.0...
regex-posix          > build
scientific           > configure
regex-posix          > Preprocessing library for regex-posix-0.96.0.0..
time-compat          > [ 1 of 20] Compiling Data.Format
regex-posix          > compiling .stack-work\dist\a3a5fe88\build\Text\Regex\Posix\Wrap_hsc_make.c failed (exit code 1)
regex-posix          > rsp file was: ".stack-work\\dist\\a3a5fe88\\build\\Text\\Regex\\Posix\\hsc4BE6.rsp"
regex-posix          > command was: C:\Users\Chase\AppData\Local\Programs\stack\x86_64-windows\ghc-8.10.2\lib\../mingw/bin\gcc.exe -c .stack-work\dist\a3a5fe88\build\Text\Regex\Posix\Wrap_hsc_make.c -o .stack-work\dist\a3a5fe88\build\Text\Regex\Posix\Wrap_hsc_make.o -D__GLASGOW_HASKELL__=810 -Dmingw32_BUILD_OS=1 -Dx86_64_BUILD_ARCH=1 -Dmingw32_HOST_OS=1 -Dx86_64_HOST_ARCH=1 -Icbits -IC:\Users\Chase\AppData\Local\Programs\stack\x86_64-windows\msys2-20180531\mingw64\include -I.stack-work\dist\a3a5fe88\build\cbits -IC:\Users\Chase\AppData\Local\Programs\stack\x86_64-windows\msys2-20180531\mingw64\include -I.stack-work\dist\a3a5fe88\build\autogen -I.stack-work\dist\a3a5fe88\build\global-autogen -include .stack-work\dist\a3a5fe88\build\autogen\cabal_macros.h -IC:\Users\Chase\AppData\Local\Programs\stack\x86_64-windows\msys2-20180531\mingw64\include -IC:\Users\Chase\AppData\Local\Programs\stack\x86_64-windows\ghc-8.10.2\lib\bytestring-0.10.10.0\include -IC:\Users\Chase\AppData\Local\Programs\stack\x86_64-windows\ghc-8.10.2\lib\base-4.14.1.0\include -IC:\Users\Chase\AppData\Local\Programs\stack\x86_64-windows\ghc-8.10.2\lib\integer-gmp-1.0.3.0\include -IC:\Users\Chase\AppData\Local\Programs\stack\x86_64-windows\ghc-8.10.2\lib/include -IC:\Users\Chase\AppData\Local\Programs\stack\x86_64-windows\ghc-8.10.2/lib/include/
regex-posix          > error: Wrap.hsc:96:10: fatal error: regex.h: No such file or directory
regex-posix          > compilation terminated.
regex-posix          >
scientific           > Configuring scientific-0.3.6.2...
time-compat          > [ 2 of 20] Compiling Data.Time.Calendar.Types
time-compat          > [ 3 of 20] Compiling Data.Time.Orphans
time-compat          > [ 4 of 20] Compiling Data.Time.Format.ISO8601.Compat
time-compat          > [ 5 of 20] Compiling Data.Time.Format.Compat
time-compat          > [ 6 of 20] Compiling Data.Time.Clock.System.Compat
scientific           > build
time-compat          > [ 7 of 20] Compiling Data.Time.Clock.POSIX.Compat
scientific           > Preprocessing library for scientific-0.3.6.2..
scientific           > Building library for scientific-0.3.6.2..
time-compat          > [ 8 of 20] Compiling Data.Time.Clock.Compat
time-compat          > [ 9 of 20] Compiling Data.Time.Calendar.Private
time-compat          > [10 of 20] Compiling Data.Time.Calendar.WeekDate.Compat
th-abstraction       > copy/register
scientific           > [1 of 5] Compiling GHC.Integer.Compat
th-abstraction       > Installing library in C:\sr\snapshots\6f4d65c6\lib\x86_64-windows-ghc-8.10.2\th-abstraction-0.4.0.0-EowzxylgIS6DLmPFoxbMpZ
time-compat          > [11 of 20] Compiling Data.Time.Calendar.Month.Compat
scientific           > [2 of 5] Compiling Utils
th-abstraction       > Registering library for th-abstraction-0.4.0.0..
scientific           > [3 of 5] Compiling Data.Scientific
time-compat          > [12 of 20] Compiling Data.Time.Calendar.Quarter.Compat
time-compat          > [13 of 20] Compiling Data.Time.Calendar.OrdinalDate.Compat
time-compat          > [14 of 20] Compiling Data.Time.Calendar.MonthDay.Compat
time-compat          > [15 of 20] Compiling Data.Time.Calendar.Easter.Compat
time-compat          > [16 of 20] Compiling Data.Time.Calendar.Compat
time-compat          > [17 of 20] Compiling Data.Time.LocalTime.Compat
time-compat          > [18 of 20] Compiling Data.Time.Compat
time-compat          > [19 of 20] Compiling Data.Time.Clock.TAI.Compat
time-compat          > [20 of 20] Compiling Data.Time.Calendar.Julian.Compat
time-compat          > copy/register
time-compat          > Installing library in C:\sr\snapshots\6f4d65c6\lib\x86_64-windows-ghc-8.10.2\time-compat-1.9.4-9MODfHpdyPK8oZKeRqK2Ff
time-compat          > Registering library for time-compat-1.9.4..
scientific           > [4 of 5] Compiling Data.Text.Lazy.Builder.Scientific
scientific           >
scientific           > src\Data\Text\Lazy\Builder\Scientific.hs:20:1: warning: [-Wunused-imports]
scientific           >     The import of `Data.Monoid' is redundant
scientific           >       except perhaps to import instances from `Data.Monoid'
scientific           > [5 of 5] Compiling Data.ByteString.Builder.Scientific
scientific           >     To import instances alone, use: import Data.Monoid()
scientific           >    |
scientific           > 20 | import Data.Monoid                  ((<>))
scientific           >    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
scientific           >
scientific           > src\Data\ByteString\Builder\Scientific.hs:25:1: warning: [-Wunused-imports]
scientific           >     The import of `Data.Monoid' is redundant
scientific           >       except perhaps to import instances from `Data.Monoid'
scientific           >     To import instances alone, use: import Data.Monoid()
scientific           >    |
scientific           > 25 | import Data.Monoid                  ((<>))
scientific           >    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
scientific           >
scientific           > copy/register
scientific           > Installing library in C:\sr\snapshots\6f4d65c6\lib\x86_64-windows-ghc-8.10.2\scientific-0.3.6.2-CuRixiYtZBNALS76xRNfDZ
scientific           > Registering library for scientific-0.3.6.2..
Progress 42/51

--  While building package regex-posix-0.96.0.0 (scroll up to its section to see the error) using:
      C:\sr\setup-exe-cache\x86_64-windows\Cabal-simple_Z6RU0evB_3.2.0.0_ghc-8.10.2.exe --builddir=.stack-work\dist\a3a5fe88 build --ghc-options " -fdiagnostics-color=always"
    Process exited with code: ExitFailure 1
```

</details>

**Comment by ryzhyk** [comment]
@Kixiron, the last commit should eliminate the dependency on `regex-posix`.

**Comment by Kixiron** [comment]
It works! Thank you so much, this will make it so much easier to develop and use!

**Comment by ryzhyk** [comment]
Closing this PR for now. I will experiment with Travis and will reopen when that's ready.

File: src/Language/DifferentialDatalog/Compile.hs [hunk 1]
```diff
@@ -481,7 +480,7 @@ tupleStruct local xs  | length xs == 0 = tuple xs
 -- structs with multiple constructor are compiled into Rust enums.
 isStructType :: Type -> Bool
 isStructType TStruct{..} | length typeCons == 1 = True
-isStructType TStruct{..}                        = False
+isStructType TStruct{}                        = False
```
Comment (mbudiu-vmw) [inline src/Language/DifferentialDatalog/Compile.hs:483]: your indentation is off

**Review by mbudiu-vmw — COMMENTED** [review]
There aren't many changes really

**Pull request closed without merging**